engine/src/main/java/org/hibernate/search/engine/search/query/spi/AbstractSearchQuery.java

```diff
@@ -54,21 +54,18 @@ public R fetch(Integer limit) {
 	}
 
 	@Override
-	public List<H> fetchHits(Integer offset, Integer limit) {
-		return fetch( offset, limit ).hits();
-	}
+	public abstract List<H> fetchHits(Integer offset, Integer limit);
 
 	@Override
 	public Optional<H> fetchSingleHit() {
 		// We don't need to fetch more than two elements to detect a problem
-		R result = fetch( 2 );
-		List<H> hits = result.hits();
-		int fetchedHitCount = result.hits().size();
+		List<H> hits = fetchHits( 2 );
+		int fetchedHitCount = hits.size();
 		if ( fetchedHitCount == 0 ) {
 			return Optional.empty();
 		}
 		else if ( fetchedHitCount > 1 ) {
-			throw log.nonSingleHit( result.totalHitCount() );
+			throw log.nonSingleHit( fetchedHitCount );
```

Comment:
This value will always be 2, while the `totalHitCount()` we used previously was actually informational. Please remove this parameter and change the log message accordingly: `Multiple hits when a single hit was expected.`.

Reply:
Good catch! Since we don't have the hit count anymore, we don't know the exact repetition number! Makes sense to be generic in the message!